### PR TITLE
Properly wait for process exit for docker helper and log errors

### PR DESCRIPTION
**Pull request by BrennanConroy**
#1648

### CODE REVIEW

File: test/Microsoft.AspNetCore.SignalR.Redis.Tests/Docker.cs [hunk 1]
```diff
@@ -125,7 +125,11 @@ private static int RunProcess(string fileName, string arguments, ILogger logger,
             process.BeginErrorReadLine();
             process.BeginOutputReadLine();
 
-            process.WaitForExit((int)timeout.TotalMilliseconds);
+            if (!process.WaitForExit((int)timeout.TotalMilliseconds))
+            {
+                process.Close();
+                logger.LogError("Closing process '{processName}' because it is running longer than the configured timeout.", fileName);
```
Comment (JamesNK) [inline test/Microsoft.AspNetCore.SignalR.Redis.Tests/Docker.cs]: Will fileName be used to replace {processName} by the logging framework? Just wondering if the string was suppose to be interpolated.

Comment (BrennanConroy) [reply test/Microsoft.AspNetCore.SignalR.Redis.Tests/Docker.cs]: Logging does in order replacement, and it's bad practice to interpolate strings passed to logging

Comment (anurse) [reply test/Microsoft.AspNetCore.SignalR.Redis.Tests/Docker.cs]: Yeah, it's intentionally not an interpolated string. Logging uses the template to generate structure log data: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?tabs=aspnetcore2x#log-message-template

Comment (JamesNK) [reply test/Microsoft.AspNetCore.SignalR.Redis.Tests/Docker.cs]: Ok. I recall a PR recently to make all the names in brackets pascal case. Should this log message follow that standard?

Comment (BrennanConroy) [reply test/Microsoft.AspNetCore.SignalR.Redis.Tests/Docker.cs]: Sure, although it doesn't really matter for test code

**Pull request merged**